### PR TITLE
fix: search not working properly in locationactivity and bookmarksfragment

### LINKED ISSUE #1427: Search is not working properly in locationactivity and bookmarksfragment.
**Actual Behaviour**

Search action not working properly .

**Expected Behaviour**

it should work properly.

**Steps to reproduce it**

Similiarly like in issue #1423 

**LogCat for the issue**

**Screenshots of the issue**

**Would you like to work on the issue?**

yes

### PR DESCRIPTION
**Pull request by amitiwary999**
Fixes #1427 

Changes: When user return from session detail activity, onResume method get call. I added code for checking search text in onResume method.

Screenshots for the change:

### CODE REVIEW

**Comment by harshitagupta30** [comment]
@amitiwary999 the search is working absolutely fine which I mentioned earlier as well I don't think we need this PR. @the-dagger @iamareebjamal Can you guys please cross check since its working fine in my device?

**Comment by iamareebjamal** [comment]
@harshitagupta30 Yes, the search is working as expected

@amitiwary999, I think you are pointing out that if we click on any session while the search box is open, and then return to the previous fragment, the results are not filtered, then yes, it is so

**Comment by harshitagupta30** [comment]
@iamareebjamal agreed. I also faced that issue. @amitiwary999 It would be great if you can clearly specify the issues and add screenshots/gif to make it clearer to understand. :+1:

**Comment by mariobehling** [comment]
@amitiwary999 Any updates?

File: android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java [hunk 1]
```diff
@@ -90,18 +90,40 @@ protected void onResume() {
...
-                    @Override
-                    public void accept(@NonNull ArrayList<Session> sessions) throws Exception {
-                        mSessions.clear();
-                        mSessions.addAll(sessions);
-                        sessionsListAdapter.notifyDataSetChanged();
-
-                        handleVisibility();
-                    }
-                }));
+        if(searchText!=null){
```
Comment (harshitagupta30) [inline android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java]: The code is not properly indented. Can you please reformat the whole condition you've written using Ctrl+Alt+L.
Also please make changes in Bookmarks fragment as well.

File: android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java [hunk 2]
```diff
@@ -90,18 +90,40 @@ protected void onResume() {
...
+                            mSessions.addAll(sessions);
+
+                            final List<Session> filteredModelList = filter(mSessions,
+                                    searchText.toLowerCase(Locale.getDefault()));
+                            Timber.tag("xyz").d("%d %d", mSessions.size(), filteredModelList.size());
+
+                            sessionsListAdapter.notifyDataSetChanged();
+                            sessionsListAdapter.animateTo(filteredModelList);
+                            sessionRecyclerView.scrollToPosition(0);
+
```
Comment (harshitagupta30) [inline android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java]: Can you please remove the extra space and commented line?

File: android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java [hunk 3]
```diff
@@ -90,18 +90,40 @@ protected void onResume() {
...
+                            mSessions.clear();
+                            mSessions.addAll(sessions);
+                            sessionsListAdapter.notifyDataSetChanged();
+
+                            handleVisibility();
+                        }
+                    }));
+        }
 
         handleVisibility();
```
Comment (harshitagupta30) [inline android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java]: Do we need to call handleVisibility() again? Please tell a reason for calling it twice. Thanks.

Comment (iamareebjamal) [inline android/app/src/main/java/org/fossasia/openevent/activities/LocationActivity.java]: @harshitagupta30 One is inside a callback, so it won't be handled immediately.

**Comment by the-dagger** [comment]
@amitiwary999 
Closing this for now, let me know when you push the changes and I'll reopen it.

**Pull request closed without merging**